internal/characters/beidou/cons.go

```diff
@@ -22,23 +23,26 @@ func (c *char) c4() {
 		if c.Core.Player.Active() != c.Index {
 			return false
 		}
-		c.Core.Status.Add("beidouc4", 600)
+		c.Core.Status.Add(c4Key, 600)
 		c.Core.Log.NewEvent("c4 triggered on damage", glog.LogCharacterEvent, c.Index).
 			Write("expiry", c.Core.F+600)
 		return false
-	}, "beidouc4")
+	}, c4Key)
+}
 
-	c.Core.Events.Subscribe(event.OnEnemyDamage, func(args ...interface{}) bool {
-		t := args[0].(combat.Target)
-		ae := args[1].(*combat.AttackEvent)
-		if ae.Info.ActorIndex != c.Index {
-			return false
-		}
-		if ae.Info.AttackTag != combat.AttackTagNormal && ae.Info.AttackTag != combat.AttackTagExtra {
-			return false
+// TODO: this should also be added to her CA
```

Comment:
alternative/addition to this change: Feel like this is something that should be available via param since we don't have player damage simmed?

IE: have a `attack[c4=1]` and `charge[c4=1]` param that will go and add the status as part of the attack routine?

Reply:
I don't feel comfortable with putting in a param for something that we could just simulate in some shape or form, see #836 and #1131. Also, we don't have params for the other on-damage procs.